Update ovirt_network.py

knumskull opened the pull request:
Sometimes it's necessary to revert back to default MTU oVirt value. This can be achieved by using '0' as input value, which was missing in documentation.

+label: docsite_pr

##### SUMMARY
<!--- Describe the change below, including rationale and design decisions -->

<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->

##### ISSUE TYPE
<!--- Pick one below and delete the rest -->
- Bugfix Pull Request
- Docs Pull Request
- Feature Pull Request
- New Module Pull Request

##### COMPONENT NAME
<!--- Write the short name of the module, plugin, task or feature below -->

##### ADDITIONAL INFORMATION
<!--- Include additional information to help people understand the change here -->
<!--- A step-by-step reproduction of the problem is helpful if there is no related issue -->

<!--- Paste verbatim command output below, e.g. before and after your change -->
```paste below

```

mattclay committed:
Use relative inventory paths instead of symlinks.

knumskull committed:
Update ovirt_network.py

Sometimes it's necessary to revert back to default MTU oVirt value. This can be achieved by using '0' as input value, which was missing in documentation.

+label: docsite_pr

knumskull committed:
Update ovirt_network.py

Sometimes it's necessary to revert back to default MTU oVirt value. This can be achieved by using '0' as input value, which was missing in documentation.

+label: docsite_pr

knumskull committed:
Update ovirt_network.py

Sometimes it's necessary to revert back to default MTU oVirt value. This can be achieved by using '0' as input value, which was missing in documentation.

+label: docsite_pr

sivel committed:
Perf graphing (#46346)

* csv of memory usage

* Fix var

* Configurable output file

* Add cpu profiling

* Valdiate the existence of cgroup files

* Add guard to prevent exception when trying to reset max memory value

* to_bytes/to_text and docs updates

* Add support for CPU results

* Just track the max, don't log all results, and then calculate max

* Restore cgroup_memory_recap, and move new functionality into cgroup_perf_recap

* Add pid count tracking, restructure to support more profilers

* Add cli tool for graphing cgroup_perf_recap data

* csv_output_dir is a path

* Correct CALLBACK_NAME

* Include uuid in csv data

* fix linting errors

* Bump version_added

* Create helper funciton to create dict from list of keys, with callable default

* Updated notes to include pids

* Print a newline after each section

* Plugin improvements

* Add option to supporess recap display
* Add default for output directory
* Add option to dictate whether or not to write files
* Add JSON-seq output option

* s/uuid/task_uuid

* Use bytes for paths

* Increase polling interval length for pids/memory

* Reduce instance attrs, change how we invoke profilers

* Shorten some line lengths

* Remove more instance attrs

* Fix some typos

* document directory creation, and catch exceptions

* Enable per task file outputs, and filename customization

* s/per_task_file/file_per_task/g

kkarolis committed:
Documentation fix for developing own plugin (#50547)

Seems like `self` is not necessary in the call args

+label: docsite_pr

mattclay committed:
Improve performance of no-tests-as-filters test.

mattclay committed:
Disable retry files for integration tests.

felixfontein committed:
docker_network IPAM test / general docker test cleanup (#50499)

* Re-enable docker_network tests.

* Basic cleanup of docker daemon.

* Add docker CLI detection.

* YAML notation.

kairoaraujo committed:
Included support to AIX group of subsystems on service module (#33043)

* Included support to AIX group subsystems

AIX systems has subsystems as services but also uses group
subsystems.
For example, spooler is a group subsystem to services
qdaemon, writesrv, and lpd.
This change enables the possibility to use also the group
susbsystmes such as spooler, nfs, etc.

When the service name is informed, first the module will
check if the name is a subsystem, if not it will check if
the name is a group subsystem and also it subsystems states.

This change makes services more flexible with AIX systems.

* Included test/legacy/aix_services.yml for tests

As discussed on IRC ansible-devel channes, was include the
legacy tests for further manual tests.

gforster committed:
Add AIX support to reboot module (#50353)

* Add ability for reboot module to work for AIX

* changelog for AIX reboot support

dagwieers committed:
team_aix: Add members to the new team (#50660)

orgito committed:
Add auth_timeout parameter when supported by paramiko (#50448)

* Add auth_timeout parameter when supported

Paramiko 2.2 introduces the auth_timeout parameter. This will set the
parameter to the same value of the timeout parameter to prevent
"Authentication timeout" errors.

* Conditionally add auth_timeout to ssh.connect

Renamed sock_kwarg to ssh_connect_kwargs and conditionally added the
auth_timeout parameter based on the installed paramiko version.

* Add changelog fragment

jborean93 committed:
win_psexec: added the pid return value to the docs (#50664)

rambleraptor committed:
New module: gcp_bigquery_table_facts (#50679)

rambleraptor committed:
New module: gcp_bigquery_dataset_facts (#50677)

jborean93 committed:
adbotmeta: add windows support to executor/powershell i (#50689)

machacekondra reviewed (approved):
Thank you! shipit

machacekondra commented on lib/ansible/modules/cloud/ovirt/ovirt_network.py:
```diff
@@ -70,6 +70,7 @@
     mtu:
         description:
             - "Maximum transmission unit (MTU) of the network."
+            - "To set oVirt default MTU, use "0" as value"
```
Oh, it fails on syntax, can you please remove double quotes around "0"?

triplepoint committed:
inventory: vagrant: rename deprectated ansible_ssh_*  (#50694)

resmo committed:
zabbix: fix author's github handle (#50700)

mindrb committed:
zabbix_host_facts: get list of templates (#50007)

Get list of templates among other items. It is useful because later one can edit this list using zabbix_host module.

rcarrillocruz committed:
Checkpoint access rule (#49937)

* WIP checkpoint_access_rule module

* Add fixes and docstrings

* Add dunder init

* Fix sanity tests issues

* Fix sanity test

* Add RETURN and EXAMPLES

* Fix example

* Fix pep8

* Add tests

* Fix pep8

* Fix pep8

dagwieers committed:
Added two new members to AIX WG

felixfontein committed:
Remove elements= from arg_spec when type is not 'list'. (#50484)

acozine committed:
fix docs for tower modules (#50710)

jainnikhil30 committed:
Fixing exception import for tower modules (#50447)

* fixing the exception import from tower modules

* Adding tests for checking tower modules are failing with correct msg

* fixed failing tests

* fixed failing test in tower_team

mkrizek committed:
tqm._initialize_notified_handlers() is noop now (#50713)

richlv committed:
Update zabbix_action.py (#50548)

* Update zabbix_action.py

Clarify when in/not in operators can be used.

Commit:
Lenovo cnos banner new module (#50145)

* Adding cnos_banner module to the list.

Commit:
Lenovo cnos l2interface (#50641)

* Adding cnos_l2 interface module and its associated Integration test as well as Unit test code

jplitza committed:
Fix sanitizing config lines (#50553)

Previously, the index got out of sync with the actual config list. Invoked with lines:
  - bad
  - first
  - bad
  - second
the sanitization would first delete index 0 and then index 2, which would result in the output
  - first
  - bad
By reversing the list, we avoid that problem (though a filter() would be nicer)

Akasurde committed:
default: callback plugin allow unreachable task to stderr (#50533)

Provide toggle flag to allow display of unreachable task to stderr
using default callback plugin.

Fixes: #48069

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

felixfontein committed:
Add delay to fix spurious test breaks.

sivel committed:
Add tests for WANT_JSON and old style modules (#50555)

* Add tests for WANT_JSON and old style modules

* quote source path

* Attempt to further appease tests

* Check for file and not just exists

* omg don't be dumb

* moar fixes

* shellcheck is the worst :)

* Test the custom modules for failure without arg files

mattclay committed:
Move any_errors_fatal test inventory into test.

mattclay committed:
Move include_import test inventory into test.

mattclay committed:
Move meta_tasks test inventory into test.

mattclay committed:
Move strategy_linear test inventory into test.

dagwieers committed:
Add mksysb module to $team_aix

Commit:
cleanup - required: false (#50648)

mattclay committed:
Merge pull request #50735 from mattclay/callback-inv

Move callback_default test inventory into test.

yuwzho committed:
azure_managed_disk enhancement (#50317)

justjais committed:
To resolve NIOS race condition when ip allocated via NIOS next available ip function (#49818)

Signed-off-by: Sumit Jaiswal <sjaiswal@redhat.com>

* resolves bug 45218

Signed-off-by: Sumit Jaiswal <sjaiswal@redhat.com>

* fixing review comments

justjais committed:
PR to fix the issue of keyerror in Nios api (#49183)

* ipv4addr keyerror fix

Signed-off-by: Sumit Jaiswal <sjaiswal@redhat.com>

* fix review comment

Signed-off-by: Sumit Jaiswal <sjaiswal@redhat.com>

rajaspachipulusu17 committed:
Pluribus Networks pn_dscp_map_pri_map module with Unit tests (#50142)

* Pluribus Networks pn_dscp_map_pri_map module with Unit tests
* Corrected method name in unit test script

yuwzho committed:
Curate the  virtual network facts module (#50311)

ikemo3 committed:
Fix: lxca_nodes misspelled in the lxca_common.py (#50734)

misspelled "requirements" to "requirement".

https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_documenting.html

tetsuyasodo committed:
fix eos_l2_interface insufficient commands (#50754)

anasbouzid committed:
schema can also be provided if type is default_privs (#50498)

+label: docsite_pr

sivel committed:
Guard creating HTTPSClientAuthHandler behind HTTPS checks (#50523)

* Guard creating HTTPSClientAuthHandler behind HTTPS checks. Fixes #50339

* linting fixup

rcarrillocruz committed:
Checkpoint host (#49936)

* Add checkpoint_host module

* Add state param to docstring

* Require ip_address if state is present

* Make else more explicit by doing and elif 404

* Remove bogus paren

* Add checkpoint_host module

rcarrillocruz committed:
Fix comma (#50766)

kbreit committed:
Clean up some of the execution of the module (#48391)

- Module now checks whether a specified network is valid very early
- Simplifies the present code flow
- No functionality enhancement

mattclay committed:
Run integration tests from temporary directory.

ci_complete

mattclay committed:
Cleanup integration test inventory. (#50753)

* Move var_blending test inventory into test.
* Remove Amazon specific inventory entry for tests.
* Remove Azure specific inventory entry for tests.
* Move var_precedence test inventory into test.
* Move unicode test inventory into test.
* Remove unused inventory entry.
* Move gathering_facts test inventory into test.
* Move delegate_to test inventory into test.
* Clean up inventory for binary_modules test.
* Clean up integration test inventory.

mattclay committed:
Compare output as JSON to avoid sort issues.

sdodsley committed:
Update purefa_volume to allow clearing of QoS limits (#50662)

sylr committed:
Azure: Add Availability Zones (#49243)

Signed-off-by: Sylvain Rabot <s.rabot@lectra.com>

yuwzho committed:
Add support for  idle_timeout, version and ipTags, curated in facts module (#48923)

dagwieers committed:
Docs: Fix highlight line (#50756)

rajaspachipulusu17 committed:
Pluribus Networks port cos bw module (#50287)

* Pluribus Networks port cos bw module

* Indentation fix

* Documentation fix

* Corrected indetentation for required_one_of

* Added maintainer name in BOTMETA.yml

* Removed maintainer name in BOTMETA.yml as my team is already there

rajaspachipulusu17 committed:
Pluribus Networks port config module (#50314)

* Pluribus Networks port config module

* Documentation fix for version added

* Doc fixes

yuwzho committed:
azure virtualmachine scaleset supports loadbalancer update (#50132)

yuwzho committed:
dns zone enhancement and return curated value (#50740)

bcoca committed:
added missing docs for option in acl module (#50775)

* added missing docs for option in acl module
* remove acl from ignore

traittinen committed:
Improve panos_commit module (#50451)

* Added description, admin commit and VSYS commit features.

* pep8 fixing

* Ansible pylint fixing + module.fail on commit fail

* Ported from lxml.etree to xml.etree. Changed username to fall back to admin if nothing else defined.

* Fixed documentation for username defaults.

sbulage committed:
Update filter documentation (#50255)

Added documentation for following filters -

* Product
* human_readable
* human_to_bytes

Signed-off-by: Satyajit Bulage <sbulage@redhat.com>

rcarrillocruz committed:
Checkpoint access rule facts (#49933)

* Add checkpoint access rule facts module

* Add docstrings and rebase with devel

* Add ANSIBLE_METADATA

rcarrillocruz committed:
Checkpoint host facts (#49935)

* Add checkpoint_host_facts module

* Rebase with devel and add docstrings

getJack committed:
Added Ansible UTM Module for managing network/interface_address objects. (#49864)

* Added Ansible UTM Module for managing network/interface_address objects.

* - Fix sanity checks
  - Fixed documentation trailing whitespaces
  - Added missing documentation

* - Fix sanity checks
  - Removed duplicate documentation

rcarrillocruz committed:
Checkpoint task facts (#49932)

* Add docstrings

* Fix pep8

rcarrillocruz committed:
Add docstrings (#49934)

jeking3 committed:
azure_rm_managed_disk: properly handle os_type on creation and update (#49385)

warusadura committed:
redfish_utils: fix reference to local variable 'systems_service' (#50298)

* fixes issue 50296

* fixes the indentation of the return statement

* Adds a conditional test into `_find_systems_resource()` to check the existence
of the Members of System resource

* updates the error message

* harden the conditional test

* Add a changelog

Akasurde committed:
Close all open filehandle (#50544)

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

Akasurde committed:
VMware: Fix vmware_vm_inventory (#50592)

* Added documentation around using vmware dynamic inventory plugin
* Fixed bug for populating host_ip in hostvars for given inventory host
* VMware: Add properties in vmware_vm_inventory

Fixes: #50249

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

ioggstream committed:
Fixes #25725: strip quotes from lsb_release and distribution description. (#31143)

hyperized committed:
Setup TLS integration test for RabbitMQ (#50363)

* Make TLS available for RabbitMQ

* Use correct path

* Include cleanup

* Also remove unused package

* Don't generate a TLS cert every time if one already has been generated

* Add newlines, clean up repository and clarify block purpose

* One too many new lines :)

* Make TLS certs static

AlanCoding committed:
Do not load user kube config if path specified (#49952)

mattclay committed:
Fix dpkg_selections integration test.

Download package from S3 since the previous location no longer exists.

felixfontein committed:
docker_volume: revert #47390 (#50663)

* Revert "Fix option change detection / force support for docker_volume. (#47390)"

This reverts commit 8ef994fbc5e0c5f996fbb5d95a9ce7c1d8658490.

* Update changelog.

* Adjusting tests.

branen committed:
Correct docsite typos: it's -> its (#50812)

mattclay committed:
Delegate ansible-test with LC_ALL=en_US.UTF-8.

This fixes encoding issues in ansible-test when running tests in
delegated environments that do not have LC_ALL or LANG set by default.

mattclay committed:
Move unrelated connection test to separate target.

Tomorrow9 committed:
VMware: add new module vmware_guest_video (#50590)

mattclay committed:
Handle non-target file deps for integration tests.

Some integration test targets have dependencies on files outside
the `test/integration/targets/` directory tree. Changes to these
dependencies can result in unexpected test failures since they do
not trigger integration tests which depend on them.

mattclay committed:
Fix ansible-test dependency processing.

Previously, the following dependencies:

A used by B
B used by C

Would have been converted to:

A used by C
B used by C

Intead of being expanded to:

A used by B
A used by C
B used by C

This change preserves the existing dependency when expanding it.

yuwzho committed:
Support curated return value for network interface and refine test (#50482)

mattclay committed:
Use different test package in pip test.

This avoids conflicts with other tests such as firewalld,
which use the decorator package installed using OS packages.

abirami-n committed:
fix_dellos6_config (#50599)

abirami-n committed:
Fix_facts (#50597)

trishnaguha committed:
Add missing parameters in get_config vyos (#50855)

Signed-off-by: Trishna Guha <trishnaguha17@gmail.com>

hannseman committed:
docker_swarm_service: Don’t add difference when update_order is None (#50655)

* Don’t add difference when update_order is None

* Add changelog fragment

rjw57 committed:
docker_swarm_service: use exact name match when finding services (#50665)

* docker_swarm_service: use exact name match when finding services

The Docker API's filtering support allows filtering for substring
matches which means that when we filter the list of running services we
may accidentally match a service called "foobar" when looking for a
service named "foo".

Fix this by filtering the list of services returned from the Docker API
so that name matches are exact. It is still worth passing the filter
parameter to the Docker API because it reduces the number of results
passed back which may be important for remote Docker connections.

Closes 50654.

* add changelog fragment for #50654

cinek810 committed:
fix use_proxy option in icinga2_host (#47671)

rcarrillocruz committed:
Add forgotten checkpoint unit tests (#50858)

I forgot to do a git add on a couple unit tests it seems

rnsc committed:
azure_rm_virtualmachine - adding support for StandardSSD_LRS (#46676)

StandardSSD_LRS have been introduced (generally available) in June 2018.
This change will add support for this.

rcarrillocruz committed:
Fix checkpoint access rule facts example (#50870)

hannseman committed:
docker_swarm_service: Documentation fixes (#50861)

* Describe labels and container_labels correctly

* Clarify reserve_memory and limit_memory docs

* Remove default from container_labels doc

* Remove trailing whitespace

* Document min api version for configs and secrets

* Add changelog fragment

* Specify type on labels and container_labels

* Consolidate required API version descriptions

* Update reserve and limit memory docs

* Use correct power-of-two units

* Remove description about limit_memory minimum 4mb

felixfontein committed:
Revert "openssl_csr: Allow to use cryptography as backend (#50324)"

This reverts commit 409f8a15bd90c87eb26ac43f1fe759346336e254.

felixfontein committed:
openssl_csr: Allow to use cryptography as backend (#50324)

* Allow to use cryptography as backend for openssl_csr.

* Use different curve.

* Adding changelog.

Includes changelog fragment for #49416, which didn't include one.

pilou- committed:
Giving felixfontein supershipit on docker stuff (#50878)

felixfontein committed:
Move changelog fragment to correct place. (#50869)

Akasurde committed:
VMware: vcenter_license fixes (#50794)

* Updated document fragment
* Used PyVmomi Helper class
* Added check for vcenter server

Fixes: #50752

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

karstenjakobsen committed:
VMware: New module vmware_drs_group (#50650)

* Added new vmware module for creating DRS VM or Host groups in a given cluster
* Fixed typo in module name
* Added better docstrings. Fixed better messaging for existing groups. Added delete example.
* Update doc
* Update lib/ansible/modules/cloud/vmware/vmware_drs_group.py
* Updated logic, so result is populated with correct data even if no changes are made
* Update lib/ansible/modules/cloud/vmware/vmware_drs_group.py
* Improved performance by getting group_obj in init()
* Fixed syntax error and added group_name as required if state is
* Added state= to integration test

Co-Authored-By: karstenjakobsen <karsten@karstenjakobsen.dk>

karstenjakobsen committed:
VMware: New module vmware_drs_group_facts (#50649)

* Added new vmware module for gathering facts for DRS groups
* Update lib/ansible/modules/cloud/vmware/vmware_drs_group_facts.py
* Updated doc
* Fixed integration test with mutual exclusive paramteres

Co-Authored-By: karstenjakobsen <karsten@karstenjakobsen.dk>

justjais committed:
Support for NIOS modules to use environment variables in provider (#49056)

* env var username/paswd support

Signed-off-by: Sumit Jaiswal <sjaiswal@redhat.com>

* including all env vars of nios doc

Signed-off-by: Sumit Jaiswal <sjaiswal@redhat.com>

* fixing default wapi version in doc-s

* fixing review comment

Signed-off-by: Sumit Jaiswal <sjaiswal@redhat.com>

* review statement fix

Signed-off-by: Sumit Jaiswal <sjaiswal@redhat.com>

jborean93 committed:
test subversion - support Ubuntu 18.04 (#50791)

mmolinac committed:
Preventing '/dev/tty not found' problems (#49950)

* Preventing '/dev/tty not found' problems when ' apt_repository` calls GPG in order to import keys.

jakeo committed:
check for chroot in systemd module (#43904)

* check for result['status'] in systemd module

* instead of checking for result['state'], actually check for chroot and warn

* allow systemctl status to work if in a chroot, update warn text

* simply change warning message

Commit:
Updated test constraints for mock assert_called() and assert_called_once()

kustodian committed:
Fix create home dir fallback (#49262)

When a user home dir is not created with `useradd`, the home dir will now
be created with umask from /etc/login.defs. Also fixed a bug in which
after a local user is deleted, and the same user exists in the central
user management system, the module would create that user's home.

mattclay committed:
Fix encoding issues with file paths. (#50830)

* Fix encoding issues with file paths.

Discovered while testing with ANSIBLE_CONFIG env var set to a path
that contained unicode characters while LC_ALL=C.

* Fix unit tests.

* Fix another path encoding issue.

horazont committed:
jabber: Fix joining group chats (#50846)

The previous code was using the legacy Group Chat 1.0 (GC1.0)
protocol to join [XEP-0045 Multi User Chats][1]. The legacy
protocol is described in § 7.2.1 of that document, the current
protocol is described in § 7.2.2.

The legacy protocol has not been in active use for more than ten
years, and servers are fading out support for it because its
presence causes issues (see for example the lengthy discussion
in [2], particularly the part starting with "A MUC
misunderstanding a presence update for a GC1.0 join").

The effect of servers fading out GC1.0 is that jabber.py cannot
send messages to rooms on servers which have done that step any
more.

This commit implements the modern join protocol, restoring
functionality. The modern join protocol is, to my knowledge,
supported by all XMPP servers which are still in use.

Prosody 0.11 is an example of a server implementation which does
not support GC1.0 anymore.

   [1]: https://xmpp.org/extensions/xep-0045.html
   [2]: https://mail.jabber.org/pipermail/standards/2017-October/033501.html

jborean93 committed:
ansible-test setup_dep_repo trust package  repo (#50795)

fabianvf committed:
Bubble up import exception content for k8s module (#50657)

* Bubble up import exception content for k8s module

Signed-off-by: Fabian von Feilitzsch <fabian@fabianism.us>

* Track down other places import exception is reported

* Add changelog fragment

jborean93 committed:
ansible-test: apt cleanup and futher 18.04 fixes (#50906)

jborean93 committed:
ansible-test: rename setup_mysql_db Ubuntu name to work with 18.04 (#50907)

jborean93 committed:
ansible-test: add postgresql var file for Ubuntu 18.04 (#50914)

mattclay committed:
Fix ansible-test target dependency issues. (#50908)

* Log dependencies at verbosity level 4.

This makes it easier to debug target dependency issues.

* Scan symlinks for target dependencies.

Some test targets use symlinks to files in other test targets.
These dependencies were previously undetected. This could result in
changes made to dependencies without triggering the dependent tests.

* Track missing target deps with `needs/target/*`.

Some existing test targets have untracked dependencies on other
test targets. This can result in changes to those dependencies
not triggering their dependent tests, resulting in test failures
after a PR is merged.

This PR adds the appropriate `needs/target/*` aliases to track
those dependencies, along with appropriate processing in
ansible-test to handle the new aliases.

* Scan meta dependencies in script targets.

Script targets are often former role targets which were converted
to allow custom invocations of ansible-playbook. These targets still
have their meta dependencies, but they were not being detected.

This could result in changes to dependencies not triggering the
targets which depend on them.

jborean93 committed:
ansible-test: fix dpkg_selections on Ubuntu 18.04 (#50915)

jborean93 committed:
ansible-test: fix cloud_init_data_facts for Ubuntu 18.04 (#50916)

jborean93 committed:
ansible-test git: kill gpg-agent to remove locks (#50917)

* ansible-test git: kill gpg-agent to remove locks

* skip gpg-agent kill for OpenSUSE

mattclay committed:
Switch set_fact test to private inventory.

Fixes a test failure when there is no explicit localhost in inventory.

rcarrillocruz committed:
Add autopublish and autoinstallpolicy behaviour to Checkpoint devices (#50862)

* Add autopublish and autoinstallpolicy behaviour to Checkpoint devices

Up till now we published and installed policy package for every operation,
however operators may not want that and only reconcile changes after a series
of changes.
Added flags to toggle this behaviour, which defaults to autopublish and
autoinstall policy package just as it was till now.
The policy package name defaults to 'standard', since it's the default one
created on the Checkpoint management server on AWS, unsure if that's common
in other setups.

* Change signature for publish and install policy

The module object is not needed

* Fix pep8

* Fix install_policy invocation

Also fix payload in publish/discard, since it seems passing the UID
when it's not needed has issues.

* Add doc fragments

* Remove default value of targets on install_policy method

It's already defaulting to None via checkpoint_arg_spec

* Fix pep8

* Remove doc fragment and push down auto options to resource modules

I realized if I put those options as doc fragments they will show up
on facts module, which do not apply, only on resource modules that
mangle with objects.

* Fix bogus param name and validate modules issues

* Fix bogus param name on checkpoint_host

dagwieers committed:
Add support for users/sites to tenants (#50924)

* Add support for users/sites to tenants

* Better work-around

ganeshrn committed:
Fix cli_config backup option check (#50926)

Fixes #50925

*  If config or rollback option is None and backup=yes in
   that case add check so edit_config() is not invoked as
   it is not required.

dagwieers committed:
Update documentation notes, copyright and license (#50927)

Akasurde committed:
Minor typos (#50371)

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

dagwieers committed:
msc_schema: Manage schemas on ACI MultiSite (#47758)

This includes the new msc_schema module and integration tests.

simon04 committed:
Consistent typesetting of "GitHub" (#50929)

resmo committed:
botmeta: adjust my cloud namespaces (#50884)

* ensure vultr tests in targets, docs covered
* add me to cloudscale namespace
* ensure cloudstack tests in targets, docs covered

resmo committed:
docs: tags: streamline indentation (#50875)

* docs: tags: streamline indentation to minimal indent

mattclay committed:
Fix false positives in no-get-exception test.

idsvandermolen committed:
gcp_compute: Add vars_prefix to prefix host_vars (#49601)

* Add vars_prefix to prefix host_vars

* Set vars_prefix default to empty string

Alb0t committed:
Aerospike migrations module (#49138)

* Push aerospike migrations module

* maxsplit only valid for python3

exception handling

bcoca committed:
Clarify clear facts (#50667)

* Revert "avoid x2 setting of set_fact when 'cacheable' (#50564)"

This reverts commit 207848f354efb556fb79664c41d426d7ba3ea4d8.

* clarify clear_facts with set_fact cacheable

 revert previous 'fix' as it will break playbooks by changing precedence
 opted to leave current behaviour but document it on both plugins to mitigate confusion

 fixes #50556

 also fix grammer, add comment, remove unused e

gundalow committed:
Revert Debug: Size of coverage file #50380 (#50945)

mattclay committed:
Add more ansible-test aliases to test docs. (#50921)

chrisedebo committed:
Update windows_setup.rst (#50941)

<!--- Your description here -->

minor spelling correction

+label: docsite_pr

mattclay committed:
Fix integration test shebangs and update tests.

* Integration tests now have their own list of allowed shebangs.
* Use `#!/usr/bin/env bash` instead of `#!/bin/bash`
  since the location is different on various platforms.

bpennypacker committed:
Ignore @bpennypacker (#50948)

dagwieers committed:
Fix relative paths

+label: docsite_pr

mattclay committed:
Further cleanup of integration test inventory. (#50919)

* Further cleanup of integration test inventory.
* Preserve aci and msc inventory in template.
* Update ansible-test inventory template handling.
* Fix classification of inventory file.

mattclay committed:
Pass ansible-test inventory path to test funcs.

This avoids declaring the inventory path/names in multiple places.
Also expose the inventory path using the INVENTORY_PATH env var.

mattclay committed:
Fix ansible-test handling of empty test targets.

Previously empty test targets were ignored by ansible-test.
This would prevent them from participating in dependency analysis.
These targets are actually empty roles, and should be processed as such.

mattclay committed:
Add missing file deps to vmware_inventory test.

trishnaguha committed:
fix ansible_connect_timeout variable in connection plugins and nxos_install_os timeout check (#50965)

Signed-off-by: Trishna Guha <trishnaguha17@gmail.com>

hannseman committed:
docker_swarm_service: Add option tests (#50699)

* Add option tests

* Check for api version on publish

* Enable configs tests

* Remove unsupported test

* Let main.yml hande creation/destruction of swarms

* Revert "Remove unsupported test"

This reverts commit 50f46bb04ef679f4235cda805188dca769ca2f72.

* Revert "Enable configs tests"

This reverts commit a94054b0bb5a3d77632bbe5aa352a8774cf93d96.

* Fix typo to properly clear secrets

rcarrillocruz committed:
Add checkpoint_object_facts module (#50933)

* Add checkpoint_object_facts module

* Fix pep8 and bogus type for ip only param

* Fix doc issues per samccann comments

rcarrillocruz committed:
Add checkpoint_session module (#50930)

* Add checkpoint_session module

* Add unit test

* Fix pep8

* Rename Checkpoint for Check Point

dagwieers committed:
Correct the command for test coverage

+label: docsite_pr

dagwieers committed:
Rename MSC modules to MSO nomenclature (#50959)

* msc_tenant: improve docs

* Rename MSC modules to MSO

* Rename MSC-related objects to MSO nomenclature

* Add missing doc fragments

rcarrillocruz committed:
Add object_type param to checkpoint_object_facts (#50982)

* Add object_type param to checkpoint_object_facts

* Add changelog fragment

* Fix sanity

* Fix sanity

* Pass type param to payload

Otherwise it does not do what is expected to do

* Add ip-only default to docstring

bcoca committed:
ensure we have required systemd env var (#50607)

* ensure we have required systemd env var
* add clarification about systemd and user scope

  fixes #50272

mattclay committed:
Fix path encoding issue loading plugins.

rambleraptor committed:
Bug fixes for GCP modules (as of 2019-01-09T15:35:45-08:00) (#50786)

gaddman committed:
Allow overwrite of SSH key (#49971)

* Allow overwrite of SSH key

* Use k: v syntax rather than k=v

bcoca committed:
Prevent duplicate role insertion into roles: (#50552)

* Corner case in which import_role would add another instance of a role with the same signature into roles: when it already existed there.

  roles:
	- name: a
  tasks:
       - import_role: name=a

  would execute role 'a' 3 times instead of the intended 2 (x2 in roles: phase +1 in tasks:)

* added tests

mattclay committed:
Use to_bytes on open and to_native on load_source.

rambleraptor committed:
Bug fixes for GCP modules (as of 2019-01-09T15:35:45-08:00) (#50785)

rambleraptor committed:
Bug fixes for GCP modules (as of 2019-01-09T15:35:45-08:00) (#50784)

rambleraptor committed:
Bug fixes for GCP modules (as of 2019-01-09T15:35:45-08:00) (#50783)

rambleraptor committed:
New GCP module: gcp_storage_object (#50684)

rambleraptor committed:
New GCP module: gcp_resourcemanager_project (#50683)

rambleraptor committed:
New GCP module: gcp_iam_service_account (#50681)

rambleraptor committed:
New GCP module: gcp_compute_interconnect_attachment (#50680)

rambleraptor committed:
New GCP module: gcp_bigquery_table (#50678)

rambleraptor committed:
New GCP module: gcp_bigquery_dataset (#50676)

mattclay committed:
Add comment explaining to_native usage.

ribeiroit committed:
Add new param "ignore_selinux_state" to seport, sefcontext, seboolean (#48945)

mattclay committed:
Fix UnicodeEncodeError for ansible --version. (#50997)

The error occurs on Python 2.x when running the following commands:

```
mkdir /tmp/Δ.cfg
ANSIBLE_CONFIG=/tmp/Δ.cfg ansible --version | tee
```

mattclay committed:
More path encoding fixes.

yungezz committed:
fix new inventory azure_rm.py (#50006)

* fix new inventory

* fix lint

* resolve comments

* resolve comments

* add defensive code

* fix response not match handler

* remove useless import

* fix lint

* add changelog

Commit:
fixed idempotence in vm extension (#51005)

vaneuk committed:
added timestamps to nxos_command module (#50261)

* added timestamps to nxos_command module

nxos_command module now returns timestamps field, which shows command execution time

* fixed unit test failure for /lib/ansible/module_utils/basic

* cosmetic changes to align with PEP 8

mattclay committed:
Remove unnecessary absolute paths from tests.

mattclay committed:
Make Shippable timing script more resilient.

This will permit the script to pass through content which could
not be previously decoded or encoded. This could occur when running
some tests on macOS using file paths with non-ASCII characters.

mkrizek committed:
Remove unnecessary calls to cache_block_tasks noop (#50715)

NilashishC committed:
Fix malformed string error for hashed_password (#51007)

* Fix malformed string error using ast

Signed-off-by: NilashishC <nilashishchakraborty8@gmail.com>

* Update doc for this option

Signed-off-by: NilashishC <nilashishchakraborty8@gmail.com>

* Revert "Update doc for this option"

This reverts commit 3579d1912129ab14d5009785ba896de820063d46.

* Revert type value

Signed-off-by: NilashishC <nilashishchakraborty8@gmail.com>

mkrizek committed:
facts: set virtualization_role for KVM hosts (#50770)

* facts: set virtualization_role for KVM hosts

Fixes #49734

* Add changelog

rcarrillocruz committed:
Add Checkpoint access layer facts module (#51016)

Commit:
Match VLAN ID as whole line instead of searching for digits in line (#51019)

Searching for digits somewhere in the output line will also match VLAN name (lines) starting with digits.

Fixes issue #50998

dagwieers committed:
Fix small typo

+label: docsite_pr

MatrixCrawler committed:
Add optional headers to utm modules (#49856)

* removed info declaration from documentation fragment as this is not implemented
* added optional headers for POST and PUT requests
* updated documentation
* added missing headers field decalaration
* removed info choice from state field
* added tests for the new utm_utils function
* fixed class invocation
* added missing required params
* fixed the pytests

jborean93 committed:
reboot - Fix connection timeout reset (#51000)

* only reset if we could retrieve the conn timeout

mattclay committed:
Encoding fixes for plugin loader and vault. (#51002)

mattclay committed:
Fix azure_rm_rediscache test resource naming.

mnecas committed:
ovirt add sync_networks (#49298)

* ovirt add sync_networks

* ovirt add network sync check

* ovirt rename check sync function

* correct python syntax ovirt host networks

* ovirt sync networks update docs

* ovirt sync network update check controll

mnecas committed:
ovirt hosts facts add cluster version filtration (#48664)

* ovirt hosts facts add cluster version filtration

* update filtering of ovirt hosts and input of cluster version

* ovirt host facts change compat version to string

jborean93 committed:
win_updates - fix category return value to be a list (#51001)

* win_updates - fix category return value to be a list

* remove unstable test alias

jborean93 committed:
setup_rabbitmq - fix erlang pinned dependencies (#51048)

dagwieers committed:
Fix parameter types and other fixes (#50111)

* Fix parameter types and other fixes

* Fix issues after review

* Fix Windows-references in system/files modules

This PR includes:
- Replacing version/v with just Ansible X.Y
- Removing Windows-alternatives from notes

* Update lib/ansible/modules/system/parted.py

Co-Authored-By: dagwieers <dag@wieers.com>

* Update lib/ansible/modules/system/service.py

Co-Authored-By: dagwieers <dag@wieers.com>

* Update lib/ansible/modules/system/service.py

Co-Authored-By: dagwieers <dag@wieers.com>

* Revert type change, move to separate PR

* Update lib/ansible/modules/files/replace.py

Co-Authored-By: dagwieers <dag@wieers.com>

* Update lib/ansible/modules/files/replace.py

Co-Authored-By: dagwieers <dag@wieers.com>

* Update lib/ansible/modules/files/replace.py

Co-Authored-By: dagwieers <dag@wieers.com>

* Update lib/ansible/modules/files/replace.py

Co-Authored-By: dagwieers <dag@wieers.com>

* Update lib/ansible/modules/files/replace.py

Co-Authored-By: dagwieers <dag@wieers.com>

* Update lib/ansible/modules/files/replace.py

Co-Authored-By: dagwieers <dag@wieers.com>

Commit:
fixed vmss required parameters handling (#51012)

ingatesystems committed:
modules: network: ingate: Add module ig_store_edit (#47675)

* modules: network: ingate: Add module ig_store_edit

* modules: network: ingate: Add module ig_config

Manage the configuration database on an Ingate SBC.

john-westcott-iv committed:
Adding TowerCLI send module (#37843)

john-westcott-iv committed:
Adding tower_workflow_launch (#42701)

john-westcott-iv committed:
Adding TowerCLI receive module (#51023)

AlanCoding committed:
Put in documented default for gcp_compute filters (#50025)

migonzalvar committed:
Make module handle its own exit (#50728)

willthames committed:
Merge module defaults groups docs into one page (#51045)

The information on module defaults groups would be
better alongside the rest of the module defaults docs

Remove the orphaned old doc

dagwieers committed:
team_solaris: Add auto-labeling for solaris (#50730)

dagwieers committed:
Add new hpux labels and $team_hpux (#51075)

Commit:
Spotinst - adding support for instance health check validation (#49211)

[src] - adding changlog fragment

aruhier committed:
Fixes and refactor netbox device module (#50672)

* Adapt netbox_device to pynetbox changes

Pynetbox API has changed, resulting in errors at the creation or
deletion of any device with the netbox_device modules. Fixes these
errors.

rebase

* Refactor netbox_device

Better error handling

Split the return values to `device` and `msg`, containing the created
device (if any), and the message to get more info about what has been
achieved (or the error if any)

milliams committed:
Fix Foreman returning host parameters (#51034)

Foreman (1.20) returns the `all_parameters` key as a list of dicts, not a dict of key-value pairs.

mattclay committed:
Enable pip integration test on RHEL 8.0.

mattclay committed:
Enable seboolean integration test on RHEL 8.0.

dagwieers committed:
team_bsd: Add auto-labeling for BSD (#51085)

We are forming an BSD Working Group with the active BSD contributors.
This relates to ansible/community#430

mattclay committed:
Enable yum integration tests on RHEL 8.0.

jctanner committed:
Fix usetty keyword in example ansible.cfg (#51084)

mattclay committed:
Fix subversion integration test on Fedora 29. (#51089)

* Fix subversion integration test on Fedora 29.

This upgrades the sqlite-libs and subversion packages to make sure
that the version of sqlite expected by subversion is installed.

* Fix compatibility with RHEL and CentOS.

bcoca committed:
allows loop pause for subseconds (#50461)

* allows loop pause for subseconds
* skip test on os x and bsd since date commands lacks nano

fixes #49951

dagwieers committed:
Welcome bcoca !

ehorning committed:
ec2_asg: doc: fix health_check_period default to match arg spec

gurch101 committed:
Add Digital Ocean Droplet API (#33984)

Create a new digital ocean droplet API to remove dopy dependency.
Also enable all digital ocean v2 parameters for droplet creation.

BondAnthony committed:
DigitalOcean module deprecation (#47272)

billdodd committed:
get standard properties for each firmware entry (#51028)

jborean93 committed:
Ansible.Become - fix 32 bit incompatibility (#51094)

Akasurde committed:
VMware: Unit test for vmware_guest (#50140)

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

jeking3 committed:
vmware_guest: improve task error handling (#49960)

ganeshrn committed:
Increase persistent command_timeout default value (#51056)

* Increase persistent command_timeout default value

*  Increase command_timeout default value from 10 to 30 sec
   to reduce frequent timeout issue for network connection
   types (netconf/network_cli/httpapi/napalm)

* Fix review comments

ganeshrn committed:
Fix connect_timeout config varaible in netconf connection (#51055)

*  Fetch connect_timeout value from get_options()
   instead of play_context

rajaspachipulusu17 committed:
Pluribus Networks switch setup module with unit test cases (#50797)

* Pluribus Networks switch setup module with unit test cases

* Module error fix and pep8 standards error fix

rajaspachipulusu17 committed:
Pluribus Networks port cos rate setting module with UT (#50856)

aaronk1 committed:
Update vmware.py minor typo (#51047)

"supported added" changed to "support added"

dagwieers committed:
New Unix maintainers (AIX, BSD, Solaris) (#51117)

tmmorin committed:
Update debug.py (#51041)

Changed "though" to "through"

jdwidari-mc committed:
Update os_quota.py fixed typo (#50981)

fixed typo, note should be not

danieljaouen committed:
homebrew_cask - add sudo_password parameter(#46471)

* Fixes #29403: homebrew_cask: Add SUDO_ASKPASS to module.run_command

* homebrew_cask: refactor sudo_password code

* homebrew_cask: more refactoring

* homebrew_cask: use %s instead of string addition

QCU87Z committed:
Update nxos_aaa_server_host.py (#49635)

* Update nxos_aaa_server_host.py

* remove host, username, and password

dagwieers committed:
More BSD/Solaris maintainers (#51138)

mattclay committed:
Remove obsolete alias from nios_txt_record test.

felixfontein committed:
openssl_csr cryptography backend, try II (#50894)

* Revert "Revert "openssl_csr: Allow to use cryptography as backend (#50324)""

This reverts commit bbd2e31e9fc278ffe74ce5e5ae0788ba90956d03.

* Remove more complicated selection copy'n'pasted from openssl_privatekey.

* Add tests for backend selection.

* Add openssl_csr test for arbitrary string commonName.

* Allow to disable commonName -> SAN copying (fixes #36690).

marcobellaccini committed:
Modified description in nmcli module doc (#50661)

* Modified description in nmcli module doc

The instructions for installing module's dependencies were incomplete.
I copied the CentOS/Fedora part from the examples section, while I figured out myself the Debian/Ubuntu package names.

+label: docsite_pr

* Fix modified description in nmcli module doc

* Fix modified description in nmcli module doc again

* Fix modified description in nmcli module doc LB

sivel committed:
Update the example for API use to reflect new context object (#50723)

hannseman committed:
docker_swarm_service: Use option_minimal_versions (#50609)

* Pass client to detect_usage function

* Use new client argument in detect_usage function

* Use option_minimal_versions

* Add min_docker_api_version

* Skip default since publish is always a list

* Add min Docker API version to module requirements

* Remove redundant keys()

* Move detect_ipvX_address_usage out of __init__

* Check for mode presence in publish ports

* Remove unused import

* Use port.get(‘mode’) to check for falsy values

* Update lib/ansible/modules/cloud/docker/docker_swarm_service.py

Co-Authored-By: hannseman <hannes@5monkeys.se>

* Add force_update to option_minimal_versions

* Detect publish mode better

* Add a trailing comma

hannseman committed:
docker_swarm_service: Fix publish idempotency when mode is None (#50882)

* Fix publish idempotency when mode is None

* Add changelog fragment

* Python 2.6 compat

* Use self.publish

* Check length of publish before comparing content

* Sort publish lists before zipping

* Enable publish tests

* python3 compat

* Don’t sort by mode as it is not safe

* Document publish suboptions and add them to args

* Add type to publish documentation

* Add choices to publish argument_spec suboptions

* Make tcp the default protocol

* Make documentation reflect protocol default

* Simplify setting mode

* Remove redundant string quoting

* Test order of publish

* Add comment about publish change detection

knumskull committed:
Merge branch 'patch-1' of https://github.com/knumskull/ansible into patch-1

Conflicts:
	lib/ansible/modules/cloud/ovirt/ovirt_network.py
